### PR TITLE
G1

### PR DESCRIPTION
**Pull request by sophieball**
added text_parser.py to the repo. 
removed comments. 
changed imports to full package name.

### CODE REVIEW

File: src/text_parser.py [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+from nltk.stem import WordNetLemmatizer
+#import mistune
+import markdown
+'''preprocess data for input'''
+
+# remove previously referenced comments
+#remove every line that starts with ">"
+def remove_reference(text):
+    pattern = r'''^>.*'''
+    result = re.sub(pattern, "", text, 0, re.MULTILINE | re.IGNORECASE | re.VERBOSE)
```
Comment (CaptainEmerson) [inline src/text_parser.py:13]: should be max 80 character width (http://google.github.io/styleguide/pyguide.html#32-line-length)

File: src/text_parser.py [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+```yml
+    staticDir:
+      - "static"
+      - "../dist"
+    ```
+'''
+def remove_markdown(text):
+    pattern = r"""
+        ^(?:\ {4}.+\n)+(?!)         # Ignore content of all indented text
+        |(^```(?:[^`]+|`(?!``))*```)  # Do not stop at single ` backticks within ```...``` blocks
```
Comment (CaptainEmerson) [inline src/text_parser.py:35]: ditto

File: src/text_parser.py [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+    return result
+
+
+def remove_url(text):
+    pattern = r'''https?://[^ ]+'''
+    result = re.sub(pattern, "", text, 0, re.MULTILINE | re.IGNORECASE | re.VERBOSE)
+    return result
+
+def remove_newline(text):
+    #text = text.encode('utf-8').replace('\n', ' ').replace('\r', '').replace('\t','')
```
Comment (CaptainEmerson) [inline src/text_parser.py:72]: rm

File: src/text_parser.py [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+    text = text.replace('\n', ' ').replace('\r', '').replace('\t', '')
```
Comment (CaptainEmerson) [inline src/text_parser.py:73]: inline?

File: src/text_parser.py [hunk 5]
```diff
@@ -0,0 +1,203 @@
...
+    return text
+
+# just remove all :
+def remove_emoji_marker(text):
+
+    pattern = r''':'''
+    text = re.sub(pattern, "", text, 0, re.MULTILINE | re.IGNORECASE | re.VERBOSE)
+    return text
+
+#has probelms. Deal with this later. Removing letters that are not supposed to be removed
```
Comment (CaptainEmerson) [inline src/text_parser.py:83]: seems contradictory

Comment (CaptainEmerson) [inline src/text_parser.py:83]: adding a TODO is fine, as long as you're explicit about problems

File: src/text_parser.py [hunk 6]
```diff
@@ -0,0 +1,203 @@
...
+
+
+# get number of tokens with non-alpha characters in the middle???
+def count_non_alpha_in_middle(text):
+    pattern = r'(^|\w)[!"#$%&\'()*+,-./:;<=>?@[\]^_`{|}~](\w|\Z)'
+    return len(re.findall(pattern, text))
+
+# get number of politeness words
+def count_politeness_words(text):
+    pass
```
Comment (CaptainEmerson) [inline src/text_parser.py:144]: rm, false sense of security

File: src/text_parser.py (hunk 1 above)
Comment (sophieball) [reply src/text_parser.py:13]: Fixed.. need to run FormatCode on all .py files..

File: src/text_parser.py (hunk 5 above)
Comment (sophieball) [reply src/text_parser.py:83]: This function is not used anywhere else.. I've deleted it now..

File: src/text_parser.py (hunk 6 above)
Comment (sophieball) [reply src/text_parser.py:144]: again, not used.. deleted. we will use convokit for counting politeness words anyway..

**Pull request merged**